Implement: Common styles for Dashboard cards

AjeshRPai opened the pull request:
## Related Issue 
Closes #18375 & #18447

## Description 
This PR creates 4 base styles for the dashboard cards and updates styles of the dashboard cards to those styles. The Styles introduced can be seen in this figma link  - YSMw2nbFAgPpjZLLbYvTfT-fi-536_3771

Also fixes the View all responses don't look tappable

## Explanation of changes 
⊕ Adds Base styles in [WordPress/src/main/res/values/dashboard_card_styles.xml](https://github.com/wordpress-mobile/WordPress-Android/pull/18377/files#diff-12d9b876fcb730b392376e60d030c0ad3e1d587eef77f40448298c92f88a072f)
⊕ [Adds: Styles for dashboard cards from the designs](https://github.com/wordpress-mobile/WordPress-Android/pull/18377/commits/d3559d78b483512d83756a71c4148756462c81a0)
 ⊜ [Updates: styling of site title in my site dashboard to use roboto](https://github.com/wordpress-mobile/WordPress-Android/pull/18377/commits/cf2d60c345cc09b254213fe152ac9307f8ec0a75)
⊜  [Updates: base style according to the material library update](https://github.com/wordpress-mobile/WordPress-Android/pull/18377/commits/7cfe4c582594baf64215dbc6f7f867d73a798e0f)
⊜  [Updates: styling of quick link ribbon items](https://github.com/wordpress-mobile/WordPress-Android/pull/18377/commits/a3ef266cae94fd1b06ef17da0637637fe228a261)

### Changes the text styles of the following Dashboard cards   
- [x] Post card 
   - [x] With Post items  
   - [x] Without Post Items 
       - [x] Create your first post card 
       - [x] Create your next post card   
   1. [Updates: Post title and subtitle](https://github.com/wordpress-mobile/WordPress-Android/pull/18377/commits/f89786bd7be7ab0bf1e2a7782e889bac58e5b12b)  
   2. [Fixes: Incorrect style reference in post card](https://github.com/wordpress-mobile/WordPress-Android/pull/18377/commits/c1f9977a036de45a156403236c15d1ca958d12b8)
- [x] Todays Stats Card - [* Updates: Stats card to inherit styles from common dashboard card style](https://github.com/wordpress-mobile/WordPress-Android/pull/18377/commits/43b118422ce075602771795396081626fe562191)
- [x] Pages Card - [Updates: style of the pages card to use dashboard card style](https://github.com/wordpress-mobile/WordPress-Android/pull/18377/commits/9bd46bccecdc71a8ce18c809a30ecc5ecfa9784c)
- [x] Activity Log Card - Using same styles as post card - [Updates: Post title and subtitle](https://github.com/wordpress-mobile/WordPress-Android/pull/18377/commits/f89786bd7be7ab0bf1e2a7782e889bac58e5b12b) 
- [x] Blogging Prompt Card - [Updates: Blaze card to inherit styles from common dashboard card style](https://github.com/wordpress-mobile/WordPress-Android/pull/18377/commits/6ab0fbbe9432ed5bfe37a0ae6c33ade0353c7f5b)
    - [x]  [Updates: Blogging Prompt card view responses text color](https://github.com/wordpress-mobile/WordPress-Android/pull/18377/commits/160cfac96e26465360ca222456fd3bc86f03fad0)
    - [x] [* Updates: Blogging Prompt card to use styles from dashboard card style](https://github.com/wordpress-mobile/WordPress-Android/pull/18377/commits/254102a84897df1b5e3d33d78282de7bc28a137c)
    - [x] The avatar size needed to be changed and since the avatar XML was being used in reader. Created a new XML just for the blogging prompt card [* Updates: Blogging Prompt card avatar size](https://github.com/wordpress-mobile/WordPress-Android/pull/18377/commits/a2121f95a56cc30e157b38cc1cca32a33e6f62e9)
- [x] Domain dashboard card - [* Updates: Quick start card & Domain dashboard card](https://github.com/wordpress-mobile/WordPress-Android/pull/18377/commits/677ee12b4b4fe9706bfc793a5ff195fc33f6a21f)
- [x] Domain Registration card - [* Updates: domain registration card subtitle style](https://github.com/wordpress-mobile/WordPress-Android/pull/18377/commits/3ab7c2211feeafc1b3fce545936c9517f3d1a3cc)
- [x] Custom Domain Card - [* Updates: styling of custom domain card title with base style](https://github.com/wordpress-mobile/WordPress-Android/pull/18377/commits/2dfd80ff0f647ee3f88b22b4a25918792a4091a9) 
- [x] Promote with Blaze card - [Updates: Blaze card to inherit styles from common dashboard card style](https://github.com/wordpress-mobile/WordPress-Android/pull/18377/commits/6ab0fbbe9432ed5bfe37a0ae6c33ade0353c7f5b)
- [x] Quick start card - [Updates: Quick start card & Domain dashboard card](https://github.com/wordpress-mobile/WordPress-Android/pull/18377/commits/677ee12b4b4fe9706bfc793a5ff195fc33f6a21f)
    - [x] Existing users quick start card 
    - [x] New users quick start card 
- [x] Plans card - [* Updates: dashboard_card_plans.xml to use common dashboard_card_styles](https://github.com/wordpress-mobile/WordPress-Android/pull/18377/commits/45e0c942bc0349a6f10cf4ef5a758fdbd99eb5f3)
- [x] Jetpack install plugin card - [Updates: the style of the Jetpack install full plugin card](https://github.com/wordpress-mobile/WordPress-Android/pull/18377/commits/c8d33f231700527c2d0e5992d5b115ab22dd10e3)


## To test:
- Check if the dashboard cards are shown as per design 
- Check if all the cards are shown as per the existing logic 
- Check if the cards behave as expected in 
   - [x] Portrait and landscape orientations.
   - [x] Light and dark modes.
   - [x] Fonts: Larger, smaller and bold text.
   - [x] High contrast.
   - [x] Talkback.
   - [x] Languages with large words or with letters/accents not frequently used in English.
   - [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
   - [x] Large and small screen sizes. (Tablet and smaller phones)

## Regression Notes
1. Potential unintended areas of impact
- Dashboard cards UI is not correct 
- Dashboard cards are not shown 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing 

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:
- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [x] High contrast.
- [x] Talkback.
- [x] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] Large and small screen sizes. (Tablet and smaller phones)
- [x] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

AjeshRPai committed:
+ Adds: Styles for dashboard cards from the designs

* Updates: Domain dashboard card with the correct style from
 dashboard card styles

AjeshRPai committed:
* [WIP] Updates: style of the pages card to use dashboard card style

- Removes: Individual Pages card pages item styles
* Refactors: to use styles from dashboard_card_styles file

AjeshRPai committed:
* [WIP] Updates: style of the pages card to use dashboard card style

- Removes: Styles of pages card footer
* Refactors: to use styles from dashboard_card_styles file

AjeshRPai committed:
* [WIP] Updates: style of the dashboard card toolbar

- Removes: Styles of dashboard card toolbar to use dashboard card style
* Updates: dashboard card styles
* Refactors: to use styles from dashboard_card_styles file

AjeshRPai committed:
- Removes: Unused resources

AjeshRPai committed:
* Updates: Footer text style with normal from Bold

AjeshRPai committed:
* Updates: Post title and subtitle

→ Moves: `MySitePostItemTitle` and `MySitePostItemSubtitle` to
 dashboard_card_styles.xml

* Updates: `MySitePostItemTitle` and `MySitePostItemSubtitle` with
parent style

AjeshRPai committed:
* Fixes: Incorrect style reference in post card

AjeshRPai committed:
* Updates: Footer style to inherit from common style

AjeshRPai committed:
* Updates: Stats card to inherit styles from common dashboard card style

AjeshRPai committed:
* Updates: Blaze card to inherit styles from common dashboard card style

- Removes: blaze card styles file - `blaze_styles.xml`

AjeshRPai committed:
* Updates: Blogging Prompt card to use styles from dashboard card style

→ Moves: Blogging Prompt card styles to dashboard_card_styles.xml
* Updates: my_site_blogging_prompt_card to use dashboard_card_styles.xml
* Updates: blogging_prompt_card_compact to use dashboard_card_styles.xml
→ Moves: Post card icon style to dashboard_card_styles.xml

osullivanchris reviewed (changes requested):
Looks great so far. Two small changes:

- On the blogging prompts card the 'View all responses' text is black(high emphasis) in the build, it should be using grey(medium emphasis) per style C. Maybe something being overridden
- I think we should reduce the size of the avatars on 'view all responses' to 24dp, as they look too big for the text next to them now the text is smaller

AjeshRPai committed:
* Updates: Blogging Prompt card view responses text color

AjeshRPai committed:
* Updates: Blogging Prompt card avatar size

+ Adds: a avatar xml for showing in card since the old one was being
used in reader as well
* Updates: the AvatarViewHolder with the new xml

AjeshRPai committed:
* Fixes: Lint errors and test failure

AjeshRPai committed:
* Updates: the styling post card when no post items are present

* Updates: the styling of create your first post card and create your
next post card - post/only one published post is present

AjeshRPai committed:
* Updates: Quick start card & Domain dashboard card

* Updates: Quick start card & Domain dashboard card to use styles from
 dashboard_card_styles.xml
+ Adds: Seperate styles for domain dashboard card in
dashboard_card_styles.xml
* Updates: domain_registration_card.xml to use the style from
 dashboard_card_styles.xml
→ Moves: Quick start card styles to dashboard_card_styles.xml
* Updates: quick_start_task_type_item.xml to use
dashboard_card_styles.xml
→ Moves: Post card icon style to dashboard_card_styles.xml

AjeshRPai committed:
Merge remote-tracking branch 'origin/issue/update-dashboard-card-styles' into issue/update-dashboard-card-styles

AjeshRPai committed:
Merge branch 'trunk' into issue/update-dashboard-card-styles

AjeshRPai committed:
* Fixes: trailing label in blogging prompt card split into two lines

AjeshRPai committed:
* Updates: dashboard_card_plans.xml to use common dashboard_card_styles

Updates: the dashboard card plans with title and description text to
use common styles from dashboard_card_styles.xml

AjeshRPai committed:
→ Moves: Error card styles to dashboard_card_styles.xml

AjeshRPai committed:
* Updates: the naming of the common styles in dashboard

AjeshRPai committed:
- Revert: local env changes

AjeshRPai committed:
Updates: Post card without post items styles to dashboard_card_styles

AjeshRPai committed:
* Updates:dashboard plans card styles to use from dashboard_card_styles

AjeshRPai committed:
* Updates: the style of the Jetpack install full plugin card

AjeshRPai committed:
* Adds: logic to show the jetpack install plugin card for review

AjeshRPai committed:
Merge branch 'trunk' into issue/update-dashboard-card-styles

AjeshRPai committed:
* Updates: domain registration card subtitle style

AjeshRPai committed:
* Updates: styling of custom domain card title with base style

AjeshRPai committed:
Revert "* Adds: logic to show the jetpack install plugin card for review"

This reverts commit d3934397ff3f379a3dc80476afd5a2808e16a3f4.

AjeshRPai committed:
- Removes: unused import

Commit:
make responses button look like a link

Commit:
add helper to get color from resource or attribute

Commit:
use the new helper to get trailing label color

Commit:
use ColorStateList instead of attribute for prompts card

AjeshRPai committed:
Merge branch 'trunk' into issue/update-dashboard-card-styles

AjeshRPai committed:
* Updates: styling of site title in my site dashboard to use roboto

AjeshRPai committed:
* Updates: base style according to the material library update

AjeshRPai committed:
* Updates: styling of my site label font weight in dashboard with

AjeshRPai committed:
* Updates: styling of quick link ribbon items

* Updates: the text size and padding values of the quick link ribbon

AjeshRPai committed:
Update checkstyle-idea.xml

AjeshRPai committed:
Update checkstyle-idea.xml

AjeshRPai committed:
* reverts: local env changes

thomashorta commented:
Regarding the Blogging Prompt card "view responses" color, we had [this issue](https://github.com/wordpress-mobile/WordPress-Android/issues/18447) open, and I submitted [this other branch](https://github.com/wordpress-mobile/WordPress-Android/tree/issue/18447-prompts-responses-link) (still didn't open a PR) to fix it. Do you think it makes sense to cherry-pick the changes from there as part of this bigger Dashboard PR?

https://github.com/wordpress-mobile/WordPress-Android/compare/issue/18447-prompts-responses-link

It differs a bit from the implementation here in the way that it adds a color state list that we use for tappable text (green with a clicked state) so tapping the text would have visual feedback. This was a request from this comment: pcdRpT-2eV-p2#comment-4631 and most likely an error when implementing the Android part, since iOS already has that in place. Does that make sense @osullivanchris?

zwarm commented on WordPress/src/main/res/layout/promote_with_blaze_card.xml:
```diff
@@ -47,20 +47,20 @@
 
             <com.google.android.material.textview.MaterialTextView
-                android:id="@+id/my_site_promote_with_blaze_card_sub_title"
+                android:id="@+id/card_detail"
```
np: It seems like the ids were following a naming pattern, so I am wondering why the name change here and not in the other instances? It seems out of place using two different patterns.

AjeshRPai replied:
Hey @zwarm , 
Good catch. This change was a mistake. I have reverted the change with the commit [8613fa0](https://github.com/wordpress-mobile/WordPress-Android/pull/18377/commits/8613fa0669ccd1b5e27d566d370dbb55cff88ec8)

zwarm reviewed (approved):
@AjeshRPai - All is looking good. I left one NP, but it's not a show stopper by any means.
This was a tricky one, thanks for wrangling it. 👏  CI is failing on a instrumentation test, I'm going to restart it, so 🤞 that by the time you are tomorrow it will be ready to merge.

@osullivanchris - nice updates!

RenanLukas reviewed (approved):
I've tested blogging prompts card, and besides @thomashorta [comment](https://github.com/wordpress-mobile/WordPress-Android/pull/18377/#issuecomment-1563054951) looks good to me :shipit: 

Thanks for taking care of this, @AjeshRPai

AjeshRPai committed:
* Reverts: the redundant change

AjeshRPai committed:
- Removes: unused imports

AjeshRPai committed:
* Fixes: unit tests

AjeshRPai commented:
Hey @thomashorta 

> Regarding the Blogging Prompt card "view responses" color, we had [this issue](https://github.com/wordpress-mobile/WordPress-Android/issues/18447) open, and I submitted [this other branch](https://github.com/wordpress-mobile/WordPress-Android/tree/issue/18447-prompts-responses-link) (still didn't open a PR) to fix it. Do you think it makes sense to cherry-pick the changes from there as part of this bigger Dashboard PR?

Thanks for pointing out these changes and giving me the commits to cherry-pick. Yep, it made sense to cherry-pick those changes. I also have updated the tests with the commit - [2261582](https://github.com/wordpress-mobile/WordPress-Android/pull/18377/commits/2261582dcb7c73ce28886046af3619eb32c5f0e0) for the changes as well. Let me know if anything is pending for the blogging prompt card. 

I have also updated the PR description to link the Blogging prompt card issue. 

cc; @RenanLukas

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/avatars/TrainOfAvatarsItem.kt:
```diff
@@ -11,7 +11,7 @@ import org.wordpress.android.ui.utils.UiString
 
 @DimenRes
-const val AVATAR_SIZE_DIMEN = R.dimen.avatar_sz_small
+const val AVATAR_SIZE_DIMEN = R.dimen.avatar_sz_extra_small
```
⚠️  This item is also used in the Reader post details, as well as the whole `TrainOfAvatarsAdapter`. Is this change expected to affect it as well? If not, I think we need to find a way of duplicating the `AvatarViewHolder` to have a specific one for Dashboard and another for Reader, and setting the correct one accordingly.

Before | After
-- | --
<img src="https://github.com/wordpress-mobile/WordPress-Android/assets/5091503/8c3377c0-3fb1-4548-b622-fb5ad2bd707e" height="600"> | <img src="https://github.com/wordpress-mobile/WordPress-Android/assets/5091503/cefbacfe-98e7-4b88-bf11-c49d2f44577a" height="600">

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/jetpackplugininstall/fullplugin/onboarding/compose/component/PluginDescription.kt:
```diff
@@ -26,10 +27,19 @@ fun PluginDescription(
+
+val dashboardCardDetailText = Typography(
+    body2 = dashboardCardDetail,
+)
```
❓ If we are using the `dashboardCardDetail` text style above, do we need this `Typography` object?

AjeshRPai replied:
Hey @thomashorta 
You are right. I have removed the `Typography` object with the commit [9bb2c09](https://github.com/wordpress-mobile/WordPress-Android/pull/18377/commits/9bb2c09b9928606a6ca63b45402c04e32751d8bc)

thomashorta commented on WordPress/src/main/res/layout/my_site_blogging_prompt_card_avatar_item.xml:
```diff
@@ -0,0 +1,18 @@
+    <ImageView
+        android:backgroundTint="?attr/colorSurface"
+        android:id="@+id/avatar"
+        style="@style/MySiteCardBloggingPromptAvatar"
```
⚠️  This makes the avatar small, but it seems it's being used in the generic "base" `AvatarViewHolder` and `TrainOfAvatarsItem` which are also used in the Reader post details. See other comments.

I think it's good to have created this new layout file with the specific dashboard needs, but I feel like we also need to duplicate the `AvatarViewHolder` and add a way for setting which ViewHolder the `TrainOfAvatarsAdapter` will use depending on the screen/component using it (Reader vs Dashboard Card).

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/avatars/AvatarViewHolder.kt:
```diff
@@ -1,7 +1,7 @@
 
 import android.view.ViewGroup
-import org.wordpress.android.databinding.AvatarItemBinding
+import org.wordpress.android.databinding.MySiteBloggingPromptCardAvatarItemBinding
```
⚠️ The `AvatarViewHolder` is used in the `TrainOfAvatarsAdapter` which is used to display post likes in the Reader detail screen. I think if we are going to have a specific `ViewHolder` for the avatar shown in the `MySiteBloggingPromptCard` then it should be a whole different `ViewHolder` that we can set the Adapter to use in that specific scenario. (see the other comment showing the changes in reader)

thomashorta reviewed (changes requested):
Hey 👋 Nice changes overall but I found a few issues related to the `TrainOfAvatar` view we use in a couple of places in the app. I am marking it as "Changes Requested" as I think the impact in the Reader details was not expected, but let me know otherwise.

I worry the code freeze date is soon so the changes I proposed may be too much / out of the scope of this PR. In that case, I would suggest reverting the size of the avatars in that component and creating a separate issue to fix it for the next release, as it doesn't seem like a major problem. Wdyt?

AjeshRPai committed:
Revert "* Updates: Blogging Prompt card avatar size"

This reverts commit a2121f95

AjeshRPai committed:
Revert "* Fixes: trailing label in blogging prompt card split into two lines"

This reverts commit 5bc5c1d555532b3fd3d80356e3d8b85b356ee146.

AjeshRPai committed:
- Removes: redundant style

AjeshRPai committed:
+ Updates: compose text style logic

AjeshRPai commented:
Hey @thomashorta 

> I worry the code freeze date is soon so the changes I proposed may be too much / out of the scope of this PR. In that case, I would suggest reverting the size of the avatars in that component and creating a separate issue to fix it for the next release, as it doesn't seem like a major problem. Wdyt?

Thanks for investigating this issue and letting me know about the areas affected. You are right. Since the code freeze date is on Monday. I have reverted all the changes related to the avatar size change in the blogging prompt card with commits [b458ee3](https://github.com/wordpress-mobile/WordPress-Android/pull/18377/commits/b458ee30c28466c33e0b947533dbefb8999f8c74), [f90113c](https://github.com/wordpress-mobile/WordPress-Android/pull/18377/commits/f90113c3effd40dcbbcaacd5915352d70e6d82cb). I will take up changing the avatar size in a different PR. 

cc: @osullivanchris

AjeshRPai committed:
Removes: redundant imports

thomashorta reviewed (approved):
The changes look good to me now. Thanks @AjeshRPai, I agree that the avatar changes are better done in a separate PR to avoid changes in scope near the code-freeze date.

Thank you for the work! 🙇
